src/go/rpk/pkg/cli/cloud/cluster/secret/create.go

```diff
@@ -0,0 +1,77 @@
+package secret
+
+import (
+	"encoding/base64"
+	"strings"
+	"unicode"
+
+	v1alpha2 "buf.build/gen/go/redpandadata/dataplane/protocolbuffers/go/redpanda/api/dataplane/v1alpha2"
+	connect "connectrpc.com/connect"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/publicapi"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func NewCreateCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var secretName, secretValue string
+
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create a new secret",
+		Long:  "Create a new secret for your Redpanda cloud cluster",
+		Run: func(cmd *cobra.Command, _ []string) {
+			// Validate secretName for non-lowercase letters
+			for _, r := range secretName {
+				if !unicode.IsLower(r) && unicode.IsLetter(r) {
+					out.Die("secret name must be lowercase")
+				}
+			}
+
+			// Load profile
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "rpk unable to load config: %v", err)
+			if p.FromCloud {
+				makeRequest := func(url string) {
+					if url == "" {
+						out.Die("URL cannot be empty")
```

Comment:
This doesn't mean anything to the end user, this will appear in the terminal and the user won't have any actionable action as `URL` is an internal variable name. 

Let's make sure we make sure at the caller that the URL is not empty, and if it's empty guide the user to the proper way of setting their profile with a Cloud cluster.

Reply:
what do you think about this error?
```go
if url == "" {
  out.Die("no cloud url found, please login with 'rpk cloud login'")
}
```

Reply:
done